announcement/announcement-tool/tool/src/test/org/sakaiproject/announcement/tool/KernelIntegrityTest.java

```diff
@@ -0,0 +1,100 @@
+package org.sakaiproject.announcement.tool;
+
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
+import org.sakaiproject.authz.api.SecurityAdvisor;
+import org.sakaiproject.authz.api.SecurityService;
+import org.sakaiproject.exception.IdInvalidException;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.exception.IdUsedException;
+import org.sakaiproject.exception.PermissionException;
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.test.ModiTest;
+import org.springframework.context.ApplicationContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import javax.inject.Inject;
+
+import static org.assertj.core.api.Assertions.*;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(name = "kernel", locations = {
+        "file:src/test/resources/mocks.xml",
+        "file:../../announcement-impl/impl/src/webapp/WEB-INF/components.xml"
+})
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class KernelIntegrityTest extends ModiTest {
```

Comment:
This is not an example, but should be moved into the kernel (without the announcement components).